Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/Batching/Batching.xaml.cs

```diff
@@ -0,0 +1,240 @@
+
+            var inputImages = GetInputImages();
+            int batchSize = GetBatchSizeFromBatchSizeComboBox();
+            CreateSessions(batchSize);
+
+            UpdateEvalText(false);
+            await Classify(inputImages);
+
+            UpdateEvalText(true);
+            await ClassifyBatched(inputImages, batchSize);
+
+            GenerateEvalResultAndUI();
+        }
+
+        private void ShowEvalUI()
+        {
+            StartInferenceBtn.IsEnabled = false;
+            EvalResults.Visibility = Visibility.Collapsed;
+            LoadingContainer.Visibility = Visibility.Visible;
+        }
+
+        private void ResetEvalMetrics()
+        {
+            avgNonBatchedDuration = 0;
+            avgBatchDuration = 0;
+        }
+
+        private List<VideoFrame> GetInputImages()
+        {
+            var birdFile = StorageFile.GetFileFromApplicationUriAsync(new Uri("ms-appx:///InputData/hummingbird.jpg")).GetAwaiter().GetResult();
+            var catFile = StorageFile.GetFileFromApplicationUriAsync(new Uri("ms-appx:///InputData/kitten.png")).GetAwaiter().GetResult();
+            var birdImage = CreateSoftwareBitmapFromStorageFile(birdFile);
+            var catImage = CreateSoftwareBitmapFromStorageFile(catFile);
+            var inputImages = new List<VideoFrame>();
+            for (int i = 0; i < 25; i++)
+            {
+                inputImages.Add(VideoFrame.CreateWithSoftwareBitmap(birdImage));
+                inputImages.Add(VideoFrame.CreateWithSoftwareBitmap(catImage));
+            }
+            return inputImages;
+        }
+
+        private SoftwareBitmap CreateSoftwareBitmapFromStorageFile(StorageFile file)
+        {
+            var stream = file.OpenAsync(FileAccessMode.Read).GetAwaiter().GetResult();
+            var decoder = BitmapDecoder.CreateAsync(stream).GetAwaiter().GetResult();
+            return decoder.GetSoftwareBitmapAsync().GetAwaiter().GetResult();
+        }
+
+        private void UpdateEvalText(bool isBatchingEval)
+        {
+            if (isBatchingEval)
+                EvalText.Text = "Inferencing Batched Inputs:";
+            else
+                EvalText.Text = "Inferencing Non-Batched Inputs:";
+        }
+
+        private void CreateSessions(int batchSizeOverride)
+        {
+            var modelPath = "ms-appx:///Models/squeezenet1.1-7.onnx";
+            nonBatchingSession_ = CreateLearningModelSession(modelPath);
+            BatchingSession_ = CreateLearningModelSession(modelPath, batchSizeOverride);
+        }
+
+        private LearningModelSession CreateLearningModelSession(string modelPath, int batchSizeOverride=-1)
+        {
+            var model = CreateLearningModel(modelPath);
+            var deviceKind =
+                (DeviceComboBox.SelectedIndex == 0) ?
+                    LearningModelDeviceKind.Cpu :
+                    LearningModelDeviceKind.DirectXHighPerformance;
+            var device = new LearningModelDevice(deviceKind);
```

Comment:
can this logic live in the DeviceComboBox class? When we go back and add support for other GPUs, or enumerating devices directly, it would be good to only have to update that one place and then all the samples get the change.

Reply:
Done.